Src/Core/Primitives/ReferenceTypeAssertions.cs

```diff
@@ -118,22 +118,40 @@ public AndConstraint<TAssertions> NotBeSameAs(TSubject unexpected, string becaus
             BeOfType(typeof(T), because, becauseArgs);
 
             return new AndWhichConstraint<TAssertions, T>((TAssertions)this, (T)(object)Subject);
-        }
-
-        /// <summary>
-        /// Asserts that the object is of the specified type <paramref name="expectedType"/>.
-        /// </summary>
-        /// <param name="expectedType">
-        /// The type that the subject is supposed to be of.
-        /// </param>
-        /// <param name="because">
-        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion 
-        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
-        /// </param>
-        /// <param name="becauseArgs">
-        /// Zero or more objects to format using the placeholders in <see cref="because" />.
-        /// </param>
-        public AndConstraint<TAssertions> BeOfType(Type expectedType, string because = "", params object[] becauseArgs)
+		}
+
+		/// <summary>
+		/// Asserts that the object is not of the specified type <typeparamref name="T"/>.
+		/// </summary>
+		/// <typeparam name="T">The type that the subject is not supposed to be of.</typeparam>
+		/// <param name="because">
+		/// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion 
+		/// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+		/// </param>
+		/// <param name="becauseArgs">
+		/// Zero or more objects to format using the placeholders in <see cref="because" />.
+		/// </param>
+		public AndWhichConstraint<TAssertions, T> NotBeOfType<T>( string because = "", params object[] becauseArgs )
+		{
+			NotBeOfType( typeof( T ), because, becauseArgs );
+
+			return new AndWhichConstraint<TAssertions, T>( (TAssertions)this, (T)(object)Subject );
```

Comment:
Sorry to comment on an already merged PR, but there is surely a copy/paste bug on this line. You have just asserted that Subject is not of type T, yet you attempt to cast it to T here. The return type of this method should probably just be AndConstraint, since you're not much wiser about the type of Subject after the test.